### ISSUE TITLE
Option -r should override --repository-file instead of being mutual exclusive

### ISSUE THREAD

**Issue by mjf** [restic/restic#3494]
```sh
restic -r /var/lib/restic/ backup /etc
Fatal: Options -r and --repository-file are mutually exclusive, please specify only one
```

The point is that I already have `RESTIC_REPOSITORY_FILE=...` declared in the environment! There is REST server (`rest-server.service`) adress there in my common setup. And now if I wanted to override it with the `-r` I could not... From my point of view this is very annoying because I have to, at least, re-source my profile or login/logout everytime I need to specify another repository. I would like to know why the two options are mutual exclusive rather than taking precedence one over another which would lead to far more flexibility and ease the usage, **IMHO**. Thanks.

**Comment by MichaelEischer** [comment]
You could either unset the environment variable, set it to an empty string, or pass `--repository-file ""` to restic. Then the options will no longer collide.

**Comment by mjf** [comment]
@MichaelEischer Yep, that's pretty clear, but it's ugly and "unergonimical". But I still don't see any point why the `-r` shouldn't override the variable... (?) I need the option to be set in the environment but there are often situations I must use the `-r` for some reason but not so often I can remember I must do some extra action such as unsetting some variable or passing some extra option. From my point of view there is some option for the **repository** and several ways to declare it: environment variable, filename containing it, and the option to set it directly. When I write for instance some scripts supporting both options and environment variables, I always carefully specify their precedence and describe it in the documentation or usage screen. It is also very common in practice. Environment variables are often used as a *sort of* static configuration (i.e. in user's login profile file etc.) or as user-defined defaults to the software. No matter where from the option value comes from it's always better to have an option to override it on demand. And this is exactly the case, IMHO...
